### PR TITLE
Juno

### PR DESCRIPTION
**Pull request by JunoNgx**
## Description

Keymap changes:
- Added community ortholinear 4x12 layout keymap `junonum`.
- Added Atom47's keymap `junonum-a47`.
- Moved `juno` from Planck and Vitamins Included to community ortholinear 4x12.
- Removed Preonic's `juno` due to poor maintenance.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by JunoNgx** [comment]
@drashna Is there any action I should take regarding this? I attempted to build this on a local clone (`vitamins_included` is coincidentally something I do own) and it succeeded.

**Comment by JunoNgx** [comment]
Understood with thanks. Layout type updated accordingly.

File: keyboards/maartenwut/atom47/keymaps/junonum-a47/keymap.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum-a47/keymap.c]: Could you please change the dash in these keymap names to an underscore?

File: layouts/community/ortho_4x12/junonum/config.h [hunk 2]
```diff
@@ -22,7 +20,7 @@
 /* enable basic MIDI features:
    - MIDI notes can be sent when in Music mode is on
 */
-
+
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/config.h]:
```suggestion

```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 3]
```diff
@@ -0,0 +1,346 @@
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+  float caps_song_on[][2] = SONG(NUM_LOCK_ON_SOUND);
+  float caps_song_off[][2] = SONG(SCROLL_LOCK_ON_SOUND);
+
+  float dpad_song_on[][2] = SONG(ROCK_A_BYE_BABY);
+  float dpad_song_off[][2] = SONG(MAJOR_SOUND);
+#endif
+
+// Disable LED on Rev6
+#ifdef KEYBOARD_planck_rev6
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+#else
+	
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        #ifdef KEYBOARD_planck_rev5
+          PORTE &= ~(1<<6);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
          writePinLow(E6);
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+        #endif
+      } else {
+        unregister_code(KC_RSFT);
+        #ifdef KEYBOARD_planck_rev5
+          PORTE |= (1<<6);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
          writePinHigh(E6);
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+      return false;
+      break;
+	  
+	  
+// Play audio upon switching Caps Lock and custom layers
+
+	case KC_CAPS:
+		if (record->event.pressed) {
+			#ifdef AUDIO_ENABLE
+			if (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
			if (host_keyboard_led_state().caps_lock) {
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+      if (clockwise) {
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]: These two can be merged into simply `tap_code(KC_PDGN)`, and the same for the `KC_PGUP`s below.

**Comment by JunoNgx** [comment]
@fauxpark Changes performed as requested.

File: keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+        layer_off(_ADJUST);
+      }
+      return false;
+      break; */
+  }
+  return true;
+}
+
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c]:
```suggestion
    if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
```

File: keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+        // Turn capslock on
+ 	   	PORTE &= ~(1 << 6);
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c]:
```suggestion
 	   	writePinLow(E6);
```

File: keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+
+    } else {
+        // Turn capslock off
+    	PORTE |= (1 << 6);
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c]:
```suggestion
    	writePinHigh(E6);
```

File: layouts/community/ortho_4x12/junonum/keymap.c [hunk 13]
```diff
@@ -0,0 +1,346 @@
...
+uint32_t layer_state_set_user(uint32_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+#else
+	
+uint32_t layer_state_set_user(uint32_t state) {
+  // LED control, lighting up when Fn layer is activated
+  state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+  switch (biton32(state)) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/junonum/keymap.c]:
```suggestion
  switch (get_highest_layer(state)) {
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 14]
```diff
@@ -0,0 +1,449 @@
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 15]
```diff
@@ -0,0 +1,449 @@
...
+  float caps_song_on[][2] = SONG(NUM_LOCK_ON_SOUND);
+  float caps_song_off[][2] = SONG(SCROLL_LOCK_ON_SOUND);
+
+  float dpad_song_on[][2] = SONG(ODE_TO_JOY);
+  float dpad_song_off[][2] = SONG(UNICODE_WINDOWS);
+#endif
+
+// Disable LED on Rev6
+#ifdef KEYBOARD_planck_rev6
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 16]
```diff
@@ -0,0 +1,449 @@
...
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+#else
+	
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 17]
```diff
@@ -0,0 +1,449 @@
...
+  // LED control, lighting up when Fn layer is activated
+  state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+  switch (biton32(state)) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
  switch (get_highest_layer(state)) {
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 18]
```diff
@@ -0,0 +1,449 @@
...
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        #ifdef KEYBOARD_planck_rev5
+          PORTE &= ~(1<<6);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
          writePinLow(E6);
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 19]
```diff
@@ -0,0 +1,449 @@
...
+        #endif
+      } else {
+        unregister_code(KC_RSFT);
+        #ifdef KEYBOARD_planck_rev5
+          PORTE |= (1<<6);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
          writePinHigh(E6);
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 20]
```diff
@@ -0,0 +1,449 @@
...
+      return false;
+      break;
+	  
+	  
+// Play audio upon switching Caps Lock and custom layers
+
+	case KC_CAPS:
+		if (record->event.pressed) {
+			#ifdef AUDIO_ENABLE
+			if (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
			if (host_keyboard_led_state().caps_lock) {
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 21]
```diff
@@ -0,0 +1,449 @@
...
+    } else {
+      if (clockwise) {
+        muse_tempo+=1;
+      } else {
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
      tap_code(KC_PGDN);
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 22]
```diff
@@ -0,0 +1,449 @@
...
+      unregister_code(KC_PGDN);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 23]
```diff
@@ -0,0 +1,449 @@
...
+    } else {
+      register_code(KC_PGUP);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
      tap_code(KC_PGUP);
```

File: layouts/community/ortho_4x12/juno/keymap.c [hunk 24]
```diff
@@ -0,0 +1,449 @@
...
+      unregister_code(KC_PGUP);
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/juno/keymap.c]:
```suggestion
```

**Comment by JunoNgx** [comment]
Sorry, did these changes last night but forgot to commit them.

**Comment by JunoNgx** [comment]
Sorry for missing out.

**Comment by JunoNgx** [comment]
@fauxpark Let me know if there's anything else I missed.

**Comment by JunoNgx** [comment]
@fauxpark Sorry, github was way more difficult to navigate than expected. Let me know if I missed anything else.

Also, tagging @zvecr for review.

File: keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c [hunk 25]
```diff
@@ -0,0 +1,152 @@
...
+#define _SPLEFT 1
+#define _SPRIGHT 2
+#define _LOWER 3
+#define _RAISE 4
+#define _ADJUST 5
+
+#define _DPAD 6
+#define _DPADNUM 7
+
+#define XXXXXX KC_NO
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c]:
```suggestion
```

File: keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c [hunk 26]
```diff
@@ -0,0 +1,152 @@
...
+	  _______,		_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,				_______,
+	  _______,					_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,	_______,	DP_ON,
+	  _______,		_______,	_______,	_______,				_______, 				_______,				_______,	_______,	_______,	_______
+	),
+	
+	[_DPAD] = LAYOUT(
+	  KC_ESC,		KC_Q,		KC_W,		KC_E,		KC_R,		KC_T,		KC_Y,		KC_U,		KC_I,		KC_O,		KC_P,		KC_DEL,		KC_BSPC,
+	  KC_TAB,		KC_A,		KC_S,		KC_D,		KC_F,		KC_G,		KC_H,		KC_J,		KC_K,		KC_L,		KC_SCLN,				KC_QUOT,
+	  KC_LSFT,					KC_Z,		KC_X,		KC_C,		KC_V,		KC_B,		KC_N,		KC_M,		KC_COMM,	KC_DOT,		KC_UP,		KC_ENT,
+	  KC_LCTL,		KC_LGUI,	KC_LALT,	XXXXXX,					DPADNUM, 				KC_SPC,					DP_OFF,		KC_LEFT,	KC_DOWN,	KC_RGHT
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/keymaps/junonum_a47/keymap.c]:
```suggestion
	  KC_LCTL,		KC_LGUI,	KC_LALT,	XXXXXXX,				DPADNUM, 				KC_SPC,					DP_OFF,		KC_LEFT,	KC_DOWN,	KC_RGHT
```

**Comment by JunoNgx** [comment]
@fauxpark Review requested.

**Comment by JunoNgx** [comment]
@fauxpark May I know what else I need for this to be merged to the main branch? Approval from @zvecr?

**Comment by fauxpark** [comment]
Not in particular, it's generally at least two collaborators.
Thanks!

**Comment by JunoNgx** [comment]
Thanks a bunch for the meticulousness and perseverance through the whole month. I did learn a lot.

**Pull request merged**